src/t_string.c

```diff
@@ -100,7 +102,32 @@ void setGenericCommand(client *c,
         if (getGenericCommand(c) == C_ERR) return;
     }
 
-    found = (lookupKeyWrite(c->db, key) != NULL);
+    const robj *value = lookupKeyWrite(c->db, key);
+    found = value != NULL;
+
+    /* Handle the IFEQ conditional check */
+    if (flags & OBJ_SET_IFEQ && found) {
+
+        if (value->type != OBJ_STRING) {
```

Comment:
Using the `checkType` method adds an error message to the client buffer and the error message will remain the same.

Reply:
```
if (!(flags & OBJ_SET_GET) && checkType(c, o, OBJ_STRING)) {
    return;
}
```

Reply:
good suggestion, thanks!